lib/charms/grafana_cloud_integrator/v0/cloud_config_requirer.py

```diff
@@ -96,52 +88,49 @@ def _events(self):
 
     @property
     def credentials(self):
-        return Credentials(
-            self._data.get("username", ""),
-            self._data.get("password", "")
-        )
+        if not all(
+            self._is_not_empty(x)
+            for x in [
+                self._data.get("username", ""),
+                self._data.get("password", ""),
+            ]):
+                return Credentials(
+                    self._data.get("username", ""),
+                    self._data.get("password", "")
+                )
+        return None
```

Comment:
Are you up for a pydantic model instead? [Ref](https://github.com/canonical/alertmanager-k8s-operator/pull/195/files#diff-72ecbe41c42d20fc2c42f2e3a84b0e5b415cdbfcb642fc3436e4e426b556bc2f).

Reply:
Yes, but I'd like to keep the scope of this PR as small as possible; feel free to open an issue for it!